odoo/addons/test_ecmascript/tests/test_ecmascript.py

```diff
@@ -0,0 +1,46 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import logging
+import os
+import re
+import subprocess
+from unittest import skipIf
+from odoo.tests.common import TransactionCase
+from odoo import tools
+from odoo.modules import get_modules, get_module_path
+
+MAX_ES_VERSION = 'es5'
+
+_logger = logging.getLogger(__name__)
+
+try:
+    es_check = tools.misc.find_in_path('es-check')
+except IOError:
+    es_check = None
+
+
+@skipIf(es_check is None, "es-check tool not found on this system")
+class TestECMAScriptVersion(TransactionCase):
+
+    longMessage = True
+
+    def test_ecmascript_version(self):
+        """ Test that there is no unsupported ecmascript in javascript files """
+
+        black_re = re.compile(r'summernote.+(intro\.js|outro.js)$')
+
+        mod_paths = [get_module_path(m) for m in get_modules()]
+        files_to_check = []
+        for p in mod_paths:
+            for dp, _, file_names in os.walk(p):
+                for fn in file_names:
+                    if fn.endswith('.js') and 'test' not in fn:
+                        files_to_check.append(os.path.join(dp, fn))
+
+        files_to_check = [f for f in files_to_check if not black_re.search(f)]
```

Comment:
Why not apply black_re while adding paths to `files_to_check` since we're already checking for the extension and a name subcomponent?